mediorum/server/serve_blob.go

```diff
@@ -308,13 +309,19 @@ func (s *MediorumServer) requireSignature(next echo.HandlerFunc) echo.HandlerFun
 				})
 			}
 
-			// check it is for this cid
-			if sig.Data.Cid != cid {
+			// check it is for this cid or id
+			if cid != "" && sig.Data.Cid != cid {
 				return c.JSON(401, map[string]string{
 					"error":  "signature contains incorrect CID",
 					"detail": fmt.Sprintf("url: %s, signature %s", cid, sig.Data.Cid),
 				})
 			}
+			if id != "" && sig.Data.ID != id {
```

Comment:
one tiny nit that's maybe worth a one-line comment: this is the jobID/upload ID here, which Discovery will get this from metadata that will be saved somewhere? (not the userId which is saved in the upload object?)